balloc/balloc.c

```diff
@@ -800,16 +813,26 @@ static int balloc_groups_write(struct m0_balloc *bal)
 	}
 	m0_mutex_init(&bgs.bgs_lock);
 	tb_cfg = (struct m0_be_tx_bulk_cfg){
-		.tbc_dom    = bal->cb_be_seg->bs_domain,
-		.tbc_datum  = &bgs,
-		.tbc_next   = &balloc_group_write_next,
-		.tbc_credit = &balloc_group_write_credit,
-		.tbc_do     = &balloc_group_write_do,
+		.tbc_q_cfg                 = {
+			.bqc_q_size_max       = 0x100,
+			.bqc_producers_nr_max = 1,
+		},
+		.tbc_workers_nr            = 0x40,
+		.tbc_partitions_nr         = 1,
+		.tbc_work_items_per_tx_max = 1,
+		.tbc_dom                   = bal->cb_be_seg->bs_domain,
+		.tbc_datum                 = &bgs,
+		.tbc_do                    = &balloc_group_write_do,
+		.tbc_done                  = &balloc_group_write_done,
 	};
 
 	rc = m0_be_tx_bulk_init(&tb, &tb_cfg);
 	if (rc == 0) {
-		M0_BE_OP_SYNC(op, m0_be_tx_bulk_run(&tb, &op));
+		m0_be_op_init(&op);
+		m0_be_tx_bulk_run(&tb, &op);
+		balloc_group_work_put(bal, &tb, &bgs);
+		m0_be_op_wait(&op);
+		m0_be_op_fini(&op);
```

Comment:
Can we replace this with:
```
M0_BE_OP_SYNC(op, ({m0_be_tx_bulk_run(&tb, &op); balloc_group_work_put(bal, &tb, &bgs);});
```
A little ugly, but it seems we can use the macro M0_BE_OP_SYNC.

Reply:
Done.